pkgs/models/default.nix

```diff
@@ -0,0 +1,30 @@
+{ lib, recurseIntoAttrs, stdenvNoCC, fetchurl, makeWrapper, sticker }:
+
+let
+  stickerModel = import ./model.nix;
+
+  # Embeddings fetcher, uses the sticker-models repository.
+  fetchEmbeddings = { name, sha256 }: {
```

Comment:
This defines a function to get an embeddings file from the sticker repo given `name` and `sha256`, `name` is used to retrieve the correct embeddings and set the `filename` which is later used in `sticker.conf`?

Reply:
Yes, just for correctness: the function does not actually get anything. Nix is a pure lazy language without side-effects, this an expression will always evaluate to the same value and does not have a side-effect such as fetching a file. Evaluating an expression, such as `fetchEmbeddings <args>` results will result in a value. Besides strings, integers, sets, etc., Some functions, such as `fetchurl` or functions defining a package return a derivation. After evaluation, such a derivation can be realized, which actually fetches what needs to be fetch and (in the case of a package builds it).

So, building a package in Nix actually consists of two steps:

1. Evaluating the package function, which results in a derivation.
2. Realizing the derivation, which results in the actual software being built.

Most `nix` utilities do both steps for you, but you can do them separately. E.g.:

```
# Evaluate the `sticker` attribute using the Nix interpreter.
% nix-instantiate -A sticker
/nix/store/fl4jrkdx6h87v4wxc0s5b5qfc2pxvja9-rust_sticker-utils-0.6.1.drv
# The .drv file is the result of evaluation (a derivation). Now, it can
# be realized, which actually builds sticker.
% nix-store -r /nix/store/fl4jrkdx6h87v4wxc0s5b5qfc2pxvja9-rust_sticker-utils-0.6.1.drv
# The preceding commands in one step.
% nix-build -A sticker
```

So, indeed, this defines a function that given a name and hash, return an attribute set with: `embeds` -> the derivation for the embeddings and `filename` -> a string containing a filename. As you point out correctly, the filename is used by the caller to update `sticker.conf`.